client/src/components/ChannelFilter.tsx

```diff
@@ -0,0 +1,64 @@
+import Select, {SingleValue} from 'react-select';
+import React from 'react';
+import styled from 'styled-components';
+import {useCallback, useEffect, useState} from 'react';
+import {fetchData} from '../services/fetch';
+
+const FilterDiv = styled.div`
+      min-width: 350px;
+      max-width: 50%;
+      flex-grow: 1;
+    `;
+
+const ChannelFilter = ({onSelected, selectedChannel}: { onSelected: CallableFunction; selectedChannel: string; }) => {
+    const [channelOptions, setChannelOptions] = useState([] as { label: string; value: string; }[]);
+    const [selectPlaceHolder, setSelectPlaceHolder] = useState('Code Reviews For All Slack Channels');
+    const handleChannelSelectionChange = useCallback((data: SingleValue<{ label: string; value: string; }>) => {
+        const { label } = channelOptions.find(({ value }) => value === data?.value) || {};
+        if (label !== selectPlaceHolder) {
+            setSelectPlaceHolder(label ?? '');
+            onSelected && onSelected(data?.value);
+        }
+    }, []);
+
+    useEffect(() => {
+        fetchData('/api/channels')
```

Comment:
The fetchData is an async function, it only return a promise.  It has to be call inside the useEffect hook to trigger the side affect when data is ready.  We can implement the code as a hook but with this small amount of code, not worth the effort.

If you move the code ouside of the useEffect, you will also need to useMemo/useState, otherwise you will end up fetching the data all the time when the component render.

Reply:
Got, you. Thanks for the clarification.